.github/ISSUE_TEMPLATE/NEW_RELEASE.md

```diff
@@ -24,23 +24,24 @@ Please do not remove items from the checklist
   - [ ] Update `RELEASE_BRANCH` and `RELEASE_VERSION` in `Makefile` and run `make prepare-release-branch`
   - [ ] Update the `CHANGELOG`
   - [ ] Submit a pull request with the changes: <!-- example #211 #214 -->
-- [ ] An OWNER [prepares a draft release](https://github.com/kubernetes-sigs/kueue/releases)
-  - [ ] Write the change log into the draft release.
-  - [ ] Run
-      `make artifacts IMAGE_REGISTRY=registry.k8s.io/kueue GIT_TAG=$VERSION`
-      to generate the artifacts and upload the files in the `artifacts` folder
-      to the draft release.
 - [ ] An OWNER creates a signed tag running
      `git tag -s $VERSION`
       and inserts the changelog into the tag description.
       To perform this step, you need [a PGP key registered on github](https://docs.github.com/en/authentication/managing-commit-signature-verification/checking-for-existing-gpg-keys).
 - [ ] An OWNER pushes the tag with
-      `git push $VERSION`
+      `git push upstream $VERSION`
   - Triggers prow to build and publish a staging container image
       `us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue:$VERSION`
+- [ ] An OWNER [prepares a draft release](https://github.com/kubernetes-sigs/kueue/releases)
+  - [ ] Create the draft release poiting out to the created tag.
+  - [ ] Write the change log into the draft release.
+  - [ ] Run
+      `make artifacts IMAGE_REGISTRY=registry.k8s.io/kueue GIT_TAG=$VERSION`
+      to generate the artifacts and upload the files in the `artifacts` folder
+      to the draft release.
```

Comment:
Add `gh release upload <tag> artifacts/*` as an optional way of uploading the files.

Reply:
In the future, we may want to add a Make target to upload the artifacts using gh command.

Reply:
Added the note about `gh release upload <tag> artifacts/*` in a separate point.